### DISCUSSION TITLE
Accessing query data types on the front end?

### DISCUSSION THREAD

**Discussion by iway1** [trpc/trpc#2235] [Q&A]
I'm trying to type some component props based on the return value from some query:

```jsx
function MyComponent() {
    const query = trpc.useQuery(['anEndpoint']);
    if(!query.data) return <Loading/>
    return (
        <Child data={query.data}/>
    )
}

function Child({
    data,
} : {
    data: ???
}) {
    //...
}
```

Any way I can type the data prop based on the return value for that prop?

**Comment by iway1** [comment]
nvm sorry this was asked before

**Reply by KrychuTM** [reply]
Where was that question answered?

**Reply by iway1** [reply]
https://trpc.io/docs/infer-types @KrychuTM
